doc/source/performance_test.rst

```diff
@@ -0,0 +1,149 @@
+================
+Performance Test
+================
+
+For better performace tweaks on specific readers, a tool ``performance_test`` under ``benchmarks`` is introduced here.
+It involves ``DASK_ARRAY__CHUNK_SIZE``, ``DASK_NUM_WORKERS``, ``OMP_NUM_THREADS`` and other options mentioned in
+:doc:`FAQ <../faq>`. This tool can loop through conditions defined by these values finally give a report in HTML.
+The report contains tables and charts for better understanding. It has two types of tests: ``simple_test``
+and ``resampler_test``.
+
+
+How it works?
+=============
+- The core is just a regular satpy script -- find datasets, load the composite, resample it if needed and
+  save it as geotiff.
+- A monitor thread using ``psutil`` records the CPU and memory usage synchronously. The sample rate is around
+  0.5 seconds. Any errors during the test will also be recorded.
+- Each round has one single condition tested. The result is stored in a csv file. After that, the machine will
+  take a 1-min rest to let the CPU cool down.
+- After all the tests finished, it collects all the result csv files, summarizing and visualizing them into the HTML
+  report.
+
+
+Preparations
+============
+1. Additional packages required
+-------------------------------
+- **psutil:** Record CPU/memory usage.
+- **pandas:** Analyze test result.
+- **matplotlib**: (Either ``matplotlib`` or ``matplotlib-base``) Plot the charts for report.
+- **py-cpuinfo**: Get the CPU model for report.
+
+
+2. Choose the composite and get corresponding datasets
+------------------------------------------------------
+Usually the composite should be the one involving a lot of computation like atmospheric correction. For most of the
+earth observing satellites, this could be ``true_color`` or something like that.
+
+Although one scene is enough to run the test, 3-5 scenes would be better to get the average.
+
+- For geostationary satellites, it is recommended to get those around **solar noon** under **full-disk** scan mode.
+- For polar orbit satellites, scenes should be around the **same area** so the intensities of the computation are similar.
+
+
+3. Organize the datasets
+------------------------
+One scene per folder. All the dataset folders should have the same naming patterns, e.g.:
+
+.. code-block:: batch
+
+    2024/06/29  09:06   <DIR>   G16_s20241691700214_e20241691709522_FLDK
+    2024/06/29  09:06   <DIR>   G16_s20241701700215_e20241701709523_FLDK
+    2024/06/29  09:06   <DIR>   G16_s20241711700217_e20241711709525_FLDK
+    2024/06/29  09:06   <DIR>   G16_s20241721700219_e20241721709527_FLDK
+    2024/06/29  09:06   <DIR>   G16_s20241731700220_e20241731709528_FLDK
+
+
+4. Do I have enough swap memory?
+--------------------------------
+Some conditions or resamplers may consume a hell of physical memory and then swap. When both are at their limits,
```

Comment:
We generally try to avoid language like this.

```suggestion
Some conditions or resamplers may consume a lot of physical memory and even use swap. When both are at their limits,
```